docs/DISTRIBUTION.md

```diff
@@ -192,12 +192,54 @@ Each run creates all hourly and daily archives for the last 14 days. To prevent
+[`AppConfigurationSigningDecorator`](/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/appconfig/structure/archive/decorator/signing/AppConfigurationSigningDecorator.java).
+
+The signing process takes the `export.bin` archive and creates a signature of the data structure which is then written into the `export.sig`.
+Please see`getSignatureFile` method located in [`SigningDecoratorOnDisk`](/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/structure/archive/decorator/signing/SigningDecoratorOnDisk.java).
+Because the `export.bin` archive contains different information (i.e keys, app configuration), there are also different
+implementations for signing each type of data structure:
+
+- [`AppConfigurationSigningDecorator`](/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/appconfig/structure/archive/decorator/signing/AppConfigurationSigningDecorator.java) - the specific code to determine how an application configuration file is supposed to be signed.
+- [`DiagnosisKeySigningDecorator`](/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/archive/decorator/signing/DiagnosisKeySigningDecorator.java) - used to sign Temporary Exposure Key files
+
+The algorithm used for signing the archives as well as other relevant information is defined in the [`application.yaml`](/services/distribution/src/main/resources/application.yaml).
 
 ## Bundling and Shifting
 
-Coming soon
+The [`DiagnosisKeyBundler`](/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/DiagnosisKeyBundler.java)
+is responsible for the core logic of the distribution service, e.g. how the diagnosis keys are grouped into date and
+hour folders on the CDN.
+
+In the preparation step, the distribution job queries all diagnosis keys that are available in the database and
+caches them into the `DiagnosisKeyBundler`. The purpose of this class is to arrange keys based on their earliest time
+(date and hour) of distribution, and provide this information to the components which create the archives.
+There are two implementations of the key bundler, loaded based on the active profile.
+It is noteworthy to mention that the productive diagnosis key bundler applies a series of policies on the keys originating
+from the country specified by the parameter`origin-country`, which is defined in the [`application.yaml`](/services/distribution/src/main/resources/application.yaml).
+Furthermore, with the parameter `apply-policies-for-all-countries` it is possible to apply the policies to all keys.
+
+Policies are as follows:
+
+- As described in the GAEN framework (more details [here](https://developer.apple.com/documentation/exposurenotification/setting_up_a_key_server)),
+diagnosis keys must not be distributed before two hours after the end of the key's expeiration window.
```

Comment:
Minor additional information, just in case you want to add it : "...after the end of the key's expeiration window (calculated by adding the rolling period of the key to its start interval number)"

Reply:
Done. 😄